src/main/java/bgu/spl/mics/Subscriber.java

```diff
@@ -17,13 +19,19 @@
  */
 public abstract class Subscriber extends RunnableSubPub {
     private boolean terminated = false;
+    // Usage of Class and Callback as without parameterized type to accommodate
+    // for different types of Messages and Callbacks
+    private ConcurrentHashMap<Class, Callback> _messageToCallback;
```

Comment:
code looks very good!
I'm not sure if this should be here or at the MessageBroker... we should refer to the assignment specification again

Reply:
This should be here, because the Message Broker doesn't care about the handling of the event. just the fact that it got to it's destination